.github/workflows/run_testing.yaml

```diff
@@ -0,0 +1,26 @@
+name: Run Unit Tests
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Run Unit Test
+    runs-on: ubuntu-latest
+    env:
+      NO_STORAGE: "true"
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install dependencies
+        run: |
+          pip install poetry
+          poetry install
```

Comment:
I admit doing the full install is a bit overkill, but it could be fine for this?

Reply:
I think it's needed actually